src/super_gradients/training/dataloaders/dataloader_factory.py

```diff
@@ -0,0 +1,110 @@
+     and
+        cfg.dataset_params.train_dataset_params or cfg.dataset_params.valid_dataset_params as defaults for DataLoader contructor.
+
+    :param dataset_params: dataset params that override the yaml configured defaults, then passed to the dataset_cls.__init__.
+    :param dataloader_params: DataLoader params that override the yaml configured defaults, then passed to the DataLoader.__init__
+    :return: DataLoader
+    """
+    GlobalHydra.instance().clear()
+    with initialize(config_path="../../recipes"):
+        # config is relative to a module
+        cfg = compose(config_name=os.path.join("dataset_params", config_name))
+
+        dataset_params = _process_dataset_params(cfg, dataset_params, train)
+
+        local_rank = get_local_rank()
+        with wait_for_the_master(local_rank):
+            dataset = dataset_cls(**dataset_params)
+
+        dataloader_params = _process_dataloader_params(cfg, dataloader_params, dataset, train)
+
+        dataloader = DataLoader(dataset=dataset, **dataloader_params)
+        return dataloader
+
+
+def _process_dataset_params(cfg, dataset_params, train):
+    default_dataset_params = cfg.dataset_params.train_dataset_params if train else cfg.dataset_params.val_dataset_params
+    default_dataset_params = hydra.utils.instantiate(default_dataset_params)
+    for key, val in default_dataset_params.items():
+        if key not in dataset_params.keys() or dataset_params[key] is None:
+            dataset_params[key] = val
+
+    return dataset_params
+
+
+def _process_dataloader_params(cfg, dataloader_params, dataset, train):
+    default_dataloader_params = cfg.dataset_params.train_dataloader_params if train else cfg.dataset_params.val_dataloader_params
+    default_dataloader_params = hydra.utils.instantiate(default_dataloader_params)
+    is_dist = super_gradients.is_distributed()
+
+    if get_param(default_dataloader_params, "sampler") is not None:
+        _instantiate_sampler(dataset, default_dataloader_params)
+    elif is_dist:
+        default_dataloader_params["sampler"] = {"DistributedSampler": {}}
+        _instantiate_sampler(dataset, default_dataloader_params)
+
+    if get_param(dataloader_params, "sampler") is not None:
+        _instantiate_sampler(dataset, dataloader_params)
+
+    _override_default_params_without_nones(dataloader_params, default_dataloader_params)
+    if get_param(dataloader_params, "batch_sampler"):
+        sampler = dataloader_params.pop("sampler")
+        batch_size = dataloader_params.pop("batch_size")
+        dataloader_params["batch_sampler"] = BatchSampler(sampler=sampler, batch_size=batch_size, drop_last=False)
+
+    return dataloader_params
+
+
+def _override_default_params_without_nones(params, default_params):
+    for key, val in default_params.items():
+        if key not in params.keys() or params[key] is None:
+            params[key] = val
```

Comment:
return value here as well

Reply:
done in https://github.com/Deci-AI/super-gradients/commit/0017eedb1ad94dbb6e06bb3c254c6c1eaf9a9ea3